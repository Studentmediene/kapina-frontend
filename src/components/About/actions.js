import {
  LOAD_ABOUT_PENDING,
  LOAD_ABOUT_SUCCESS,
  LOAD_ABOUT_FAILED,
} from './constants';

export function loadAbout() {
  return {
    type: LOAD_ABOUT_PENDING,
  };
}

export function aboutLoaded(about) {
  return {
    type: LOAD_ABOUT_SUCCESS,
    about,
  };
}

export function aboutError() {
  return {
    type: LOAD_ABOUT_FAILED,
  };
}
