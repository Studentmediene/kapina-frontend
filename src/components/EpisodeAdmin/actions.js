/*
 *
 * EpisodeAdmin actions
 *
 */
import {
  ADD_EPISODE_PENDING,
  ADD_EPISODE_SUCCESS,
  ADD_EPISODE_FAILED,
  LOAD_DIGAS_EPISODES_PENDING,
  LOAD_DIGAS_EPISODES_SUCCESS,
  LOAD_DIGAS_EPISODES_FAILED,
  CLEAR_DIGAS_EPISODES,
} from './constants';

export function addEpisodePending(episode) {
  return {
    type: ADD_EPISODE_PENDING,
    episode,
  };
}

export function addEpisodeSuccess() {
  return {
    type: ADD_EPISODE_SUCCESS,
  };
}

export function addEpisodeError(error) {
  return {
    type: ADD_EPISODE_FAILED,
    error,
  };
}

export function loadDigasEpisodesPending(digasId) {
  return {
    type: LOAD_DIGAS_EPISODES_PENDING,
    digasId,
  };
}

export function loadDigasEpisodesSuccess(digasEpisodes) {
  return {
    type: LOAD_DIGAS_EPISODES_SUCCESS,
    digasEpisodes,
  };
}

export function loadDigasEpisodesError(error) {
  return {
    type: LOAD_DIGAS_EPISODES_FAILED,
    error,
  };
}

export function clearDigasEpisodes() {
  return {
    type: CLEAR_DIGAS_EPISODES,
  };
}
