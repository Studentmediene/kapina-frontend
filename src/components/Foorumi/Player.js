/* eslint-disable */
export const PlayerState = Object.freeze({
  STANDING: 'STANDING',
  RUNNING: 'RUNNING',
  JUMPING: 'JUMPING',
  THROWING: 'THROWING',
});



export default class Player {
  constructor(context, width, height, stageWidth, stageHeight, image, flippedImage) {
    this.context = context;
    this.width = width;
    this.height = height;
    this.stageWidth = stageWidth;
    this.stageHeight = stageHeight;
    this.image = image;
    this.flippedImage = flippedImage;
  }

  // Indexes used to render the correct sprite on canvas
  standIndex = 0;
  runStartIndex = 1;
  runEndIndex = 8;
  throwStartIndex = 9;
  throwEndIndex = 11;
  jumpStartIndex = 12;
  jumpEndIndex = 13;

  // Fields containing the current state of sprite
  frameIndex = 0;
  tickCount = 0;
  ticksPerFrame = 5;
  numberOfFrames = 14;
  spriteWidth = 96;
  spriteHeight = 80;

  // Fields containing the state of the player
  x = 0; // The centre of the player
  y = 0; // The centre of the player
  currentSpeedX = 8;
  currentSpeedY = -40;

  hitBoxWidth = 40;
  hitBoxHeight = 80;


  state = PlayerState.STANDING;


  render = function () {
    let image = this.image;
    if (this.isFlipped) {
      image = this.flippedImage;
    }
    this.context.drawImage(
      image,
      this.frameIndex * this.width / this.numberOfFrames, // Start clip x of image
      0, // Start y clip of image
      this.width / this.numberOfFrames, // End x clip of image
      this.height, // End y clip of image
      this.x - this.spriteWidth/2, // Start x on canvas
      this.y, // Start y on canvas
      this.width / this.numberOfFrames, // Width on canvas
      this.height // Height on canvas
    );
    /*this.context.strokeRect(
      this.x - this.hitBoxWidth/2,
      this.y,
      this.hitBoxWidth,
      this.hitBoxHeight);*/
  };

  update(keystate, windowOffset, maxWindowOffset) {

    /**
     * Must be run after this._updateWindowOffset();
    */
    if(keystate[37]){ // Pressing left
      this.isFlipped = true;
      // If we are not jumping, then we are running
      if (this.state !== PlayerState.JUMPING) {
        this.state = PlayerState.RUNNING;
      }
      if (windowOffset === 0) {
        // We are at the left edge
        //console.log('We are moving left, and are at the left edge')
        this.x -= this.currentSpeedX;
      } else if (windowOffset === maxWindowOffset) {
        // We are at the right edge
        //console.log('We are moving left and are at the right edge')
        this.x -= this.currentSpeedX;
      }
    } else if(keystate[39]){ // Pressing right
      this.isFlipped = false;
      // If we are not jumping, then we are running
      if (this.state !== PlayerState.JUMPING) {
        this.state = PlayerState.RUNNING;
      }
      if (windowOffset === 0) {
        // We are at the left edge
        //console.log('We are moving right, and are at the left edge')
        this.x += this.currentSpeedX
      } else if (windowOffset === maxWindowOffset) {
          // We are at the right edge
          //console.log('We are moving right and are at the right edge')
          this.x += this.currentSpeedX
      }
    } else if(this.state !== PlayerState.JUMPING) {
      this.state = PlayerState.STANDING;
    }

    // Handle jumping
    // If the player is not already jumping, and "up" is pressed...
    if(this.state !== PlayerState.JUMPING && keystate[38]) {
      this.state = PlayerState.JUMPING;
    }
    if(this.state === PlayerState.JUMPING) {
      this.y += this.currentSpeedY;
      this.currentSpeedY += 4;
      if(this.y > this.stageHeight - 37*2 - this.hitBoxHeight/2){
        this.state = PlayerState.STANDING;
        this.y = this.stageHeight - 37*2 - this.hitBoxHeight/2;
        this.currentSpeedY = -40;
      }
    }

    // Handle cases where the player is about to leave the stage
    if(this.x < this.hitBoxWidth/2){
      console.log('Player hit left wall!')
      this.x = this.hitBoxWidth/2;
    } else if (this.x > this.stageWidth - this.hitBoxWidth/2) {
      console.log('Player hit right wall!')
      this.x = this.stageWidth - this.hitBoxWidth/2;
    }

    // Handle cases where the player runs past the middle of the screen
    if(windowOffset === 0 && this.x > this.stageWidth/2){
      // We are at the left edge
      console.log('Player ran in the right direction across the middle!')
      this.x = this.stageWidth/2
    } else if (windowOffset === maxWindowOffset && this.x < this.stageWidth/2) {
      // We are at the right edge
      console.log('Player ran left across the middle!')
      this.x = this.stageWidth/2;
    }


    if (this.state === PlayerState.RUNNING) {
      this.updateRun()
    } else if (this.state === PlayerState.JUMPING) {
      this.updateJump()
    } else if (this.state === PlayerState.THROWING) {
      this.updateThrow()
    } else {
      this.updateStand()
    }
  }


  _update = function (startIndex, endIndex) {
    if (!this.isFlipped) {
        if (this.frameIndex < startIndex || this.frameIndex > endIndex) {
          this.frameIndex = startIndex;
        }
        this.tickCount += 1;
        if (this.tickCount > this.ticksPerFrame) {
          this.tickCount = 0;
          if (this.frameIndex < endIndex) {
            this.frameIndex += 1;
          }
          else {
            this.frameIndex = startIndex;
          }
        }
    }
    else {
      /* Adjust the indexes to the flipped image */
      startIndex = this.numberOfFrames - startIndex - 1;
      endIndex = this.numberOfFrames - endIndex - 1;
      //console.log(startIndex, endIndex);
      if (this.frameIndex > startIndex || this.frameIndex < endIndex) {
        this.frameIndex = startIndex;
      }
      this.tickCount += 1;
      if (this.tickCount > this.ticksPerFrame) {
        this.tickCount = 0;
        if (this.frameIndex > endIndex) {
          this.frameIndex -= 1;
        }
        else {
          this.frameIndex = startIndex;
        }
      }
    }

  }


  /*
    Cycles through the running sprites.
  */
  updateRun = function () {
    this._update(this.runStartIndex, this.runEndIndex);
  };

  /*
    Chooses the index for the stand sprite to be the active frameindex.
  */
  updateStand = function () {
    if (this.isFlipped) {
      this.frameIndex = this.numberOfFrames - this.standIndex - 1;
    }
    else {
      this.frameIndex = this.standIndex;
    }
  }


  /*
    Cycles through the throwing sprites.
  */
  updateThrow = function() {
    this._update(this.throwStartIndex, this.throwEndIndex);
  }

  updateJump = function() {
    this._update(this.jumpStartIndex, this.jumpEndIndex);
  }

}
