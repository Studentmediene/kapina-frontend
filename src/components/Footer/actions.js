import {
  LOAD_FOOTER_PENDING,
  LOAD_FOOTER_SUCCESS,
  LOAD_FOOTER_FAILED,
} from './constants';

export function loadFooter() {
  return {
    type: LOAD_FOOTER_PENDING,
  };
}

export function footerLoaded({ chiefEditor, radioEditor }) {
  return {
    type: LOAD_FOOTER_SUCCESS,
    chiefEditor,
    radioEditor,
  };
}

export function footerError() {
  return {
    type: LOAD_FOOTER_FAILED,
  };
}
