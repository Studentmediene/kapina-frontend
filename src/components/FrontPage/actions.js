/*
 *
 * FrontPage actions
 *
 */

import {
  LOAD_FRONT_PAGE_POSTS_PENDING,
  LOAD_FRONT_PAGE_POSTS_SUCCESS,
  LOAD_FRONT_PAGE_POSTS_FAILED,
} from './constants';

export function loadFrontPagePosts(pageNumber) {
  return {
    type: LOAD_FRONT_PAGE_POSTS_PENDING,
    pageNumber,
  };
}

export function frontPagePostsLoaded(posts) {
  return {
    type: LOAD_FRONT_PAGE_POSTS_SUCCESS,
    posts,
  };
}

export function frontPagePostsError() {
  return {
    type: LOAD_FRONT_PAGE_POSTS_FAILED,
  };
}
