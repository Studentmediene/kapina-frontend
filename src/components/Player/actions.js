/*
 *
 * Player actions
 *
 */

import {
  PLAY_LIVE,
  PLAY_LIVE_PENDING,
  GET_PODCAST_PLAYLIST_PENDING,
  GET_PODCAST_PLAYLIST_SUCCESS,
  GET_PODCAST_PLAYLIST_FAIELD,
  GET_ON_DEMAND_PLAYLIST_PENDING,
  GET_ON_DEMAND_PLAYLIST_SUCCESS,
  GET_ON_DEMAND_PLAYLIST_FAILED,
  GET_LIVE_TITLE,
  LIVE_TITLE_UPDATER,
  TOGGLE_PLAY_PAUSE,
  RESUME,
  PAUSE,
  PLAYER_STATUS,
  PLAY_NEXT,
  PLAY_PREVIOUS,
  PLAY_ON_DEMAND_EPISODE,
} from './constants';

export function playLive(offset = 0) {
  return {
    type: PLAY_LIVE_PENDING,
    offset,
  };
}

export function playLiveURL(url, offset = 0) {
  return {
    type: PLAY_LIVE,
    url,
    offset,
  };
}

export function getPodcastPlaylist(episodeId, offset = 0) {
  return {
    type: GET_PODCAST_PLAYLIST_PENDING,
    episodeId,
    offset,
  };
}

export function podcastPlaylistLoaded(playlist, index, offset = 0) {
  return {
    type: GET_PODCAST_PLAYLIST_SUCCESS,
    playlist,
    index,
    offset,
  };
}

export function podcastPlaylistError() {
  return {
    type: GET_PODCAST_PLAYLIST_FAIELD,
  };
}

export function getOnDemandPlaylist(episodeId, offset = 0) {
  return {
    type: GET_ON_DEMAND_PLAYLIST_PENDING,
    episodeId,
    offset,
  };
}

export function onDemandPlaylistLoaded(playlist) {
  return {
    type: GET_ON_DEMAND_PLAYLIST_SUCCESS,
    playlist,
  };
}

export function onDemandPlaylistError() {
  return {
    type: GET_ON_DEMAND_PLAYLIST_FAILED,
  };
}

export function currentShowTitle(liveTitle) {
  return {
    type: GET_LIVE_TITLE,
    liveTitle,
  };
}

export function togglePlayPause() {
  return {
    type: TOGGLE_PLAY_PAUSE,
  };
}

export function resume() {
  return {
    type: RESUME,
  };
}

export function pause() {
  return {
    type: PAUSE,
  };
}

export function playerStatus({ paused }) {
  return {
    type: PLAYER_STATUS,
    paused,
  };
}

export function playNext() {
  return {
    type: PLAY_NEXT,
  };
}

export function playPrevious() {
  return {
    type: PLAY_PREVIOUS,
  };
}

export function playOnDemandEpisode(index, offset = 0) {
  return {
    type: PLAY_ON_DEMAND_EPISODE,
    index,
    offset,
  };
}

export function liveTitleUpdater() {
  return {
    type: LIVE_TITLE_UPDATER,
  };
}
