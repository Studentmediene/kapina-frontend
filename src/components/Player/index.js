/*eslint no-console: ["error", { allow: ["log"] }] */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { soundManager } from 'soundmanager2';

import AudioProgress from './components/AudioProgress';
import AudioControls from './components/AudioControls';
import SoundManager from './components/SoundManager';
import {
  playLive,
  getPodcastPlaylist,
  getOnDemandPlaylist,
  pause,
  resume,
  togglePlayPause,
  playNext,
  playPrevious,
} from './actions';
import {
  selectPlaylist,
  selectIndex,
  selectOffset,
  selectLive,
  selectPlayingTitle,
  selectPaused,
  selectUrl,
} from './selectors';
import styles from './styles.css';

class Player extends React.Component {
  constructor(props) {
    super(props);
    // Initial volume
    this.volume = 80;
  }

  state = {
    // Number of seconds played
    position: 0,
    // Duration of the audio (estimate)
    duration: 0,
  };

  componentWillMount() {
    soundManager.onready(() => {
      // Load correct URL based on browser support
      const oggUrl = 'https://direkte.radiorevolt.no/revolt.ogg';
      const aacUrl = 'https://direkte.radiorevolt.no/revolt.aac';
      if (soundManager.canPlayURL(oggUrl)) {
        this.liveUrl = oggUrl;
      } else {
        this.liveUrl = aacUrl;
      }
    });
  }

  resetPosition() {
    this.setState({
      position: 0,
    });
  }

  onSeek(seekPosition) {
    this.setState({
      position: seekPosition,
    });
  }

  whilePlaying(soundObject) {
    this.setState({
      position: soundObject.position,
      durationEstimate: soundObject.durationEstimate,
    });
  }

  render() {
    const { position } = this.state;

    return (
      <div className={styles.container} title={this.props.playingTitle}>
        <SoundManager
          url={this.props.url}
          paused={this.props.paused}
          position={position}
          volume={this.volume}
          whilePlaying={(...a) => this.whilePlaying(...a)}
          onPlay={() => {
            console.log('onPlay event');
          }}
          onPause={() => {
            this.props.pause();
          }}
          onResume={() => {
            this.props.resume();
          }}
          onFinishedPlaying={() => {
            this.resetPosition();
            this.props.playNext();
          }}
        />
        <AudioControls
          playNext={() => this.props.playNext()}
          playPrevious={() => {
            if (!this.props.live) {
              const backLimit = 2 * 1000; // two seconds
              if (this.state.position >= backLimit) {
                this.resetPosition();
              } else {
                this.props.playPrevious();
              }
            }
          }}
          togglePlayPause={() => this.props.togglePlayPause()}
          paused={this.props.paused}
        />
        <AudioProgress
          displayText={this.props.playingTitle}
          live={this.props.live}
          paused={this.props.paused}
          url={this.props.url}
          position={this.state.position}
          durationEstimate={this.state.durationEstimate}
          onSeek={position => this.onSeek(position)}
        />
      </div>
    );
  }
}

Player.propTypes = {
  playLive: PropTypes.func,
  playPodcast: PropTypes.func,
  playOnDemand: PropTypes.func,
  playlist: PropTypes.oneOfType([PropTypes.bool, PropTypes.array]),
  offset: PropTypes.number,
  index: PropTypes.number,
  live: PropTypes.bool,
  paused: PropTypes.bool,
  url: PropTypes.string,
  playingTitle: PropTypes.string,
  togglePlayPause: PropTypes.func.isRequired,
  resume: PropTypes.func.isRequired,
  pause: PropTypes.func.isRequired,
  playNext: PropTypes.func.isRequired,
  playPrevious: PropTypes.func.isRequired,
};

Player.defaultProps = {
  paused: true,
  url: null,
};

const mapStateToProps = createStructuredSelector({
  playlist: selectPlaylist(),
  live: selectLive(),
  offset: selectOffset(),
  index: selectIndex(),
  paused: selectPaused(),
  url: selectUrl(),
  playingTitle: selectPlayingTitle(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    playLive: (offset = 0) => dispatch(playLive(offset)),
    playPodcast: (episodeId, offset = 0) =>
      dispatch(getPodcastPlaylist(episodeId, offset)),
    playOnDemand: (episodeId, offset = 0) =>
      dispatch(getOnDemandPlaylist(episodeId, offset)),
    togglePlayPause: () => dispatch(togglePlayPause()),
    resume: () => dispatch(resume()),
    pause: () => dispatch(pause()),
    playNext: () => dispatch(playNext()),
    playPrevious: () => dispatch(playPrevious()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Player);
