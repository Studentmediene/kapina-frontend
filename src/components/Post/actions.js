/*
 *
 * Post actions
 *
 */

import {
  LOAD_POST_PENDING,
  LOAD_POST_SUCCESS,
  LOAD_POST_FAILED,
} from './constants';

export function loadPost(slug) {
  return {
    type: LOAD_POST_PENDING,
    slug,
  };
}

export function postLoaded(post, slug) {
  return {
    type: LOAD_POST_SUCCESS,
    post,
    slug,
  };
}

export function postError() {
  return {
    type: LOAD_POST_FAILED,
  };
}
