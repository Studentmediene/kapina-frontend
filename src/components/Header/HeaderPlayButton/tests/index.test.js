import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import ConnectedHeaderPlayButton, { HeaderPlayButton } from '../';
import { playLive } from 'components/Player/actions';
import { fromJS } from 'immutable';

describe('<HeaderPlayButton />', () => {
  it('renders correctly', () => {
    const mockProps = {
      playLive: jest.fn(),
    };
    const tree = shallow(<HeaderPlayButton {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('calls playLive when clicked', () => {
    const mockProps = {
      playLive: jest.fn(),
    };
    const tree = mount(<HeaderPlayButton {...mockProps} />);
    tree.find('.playButton').simulate('click');
    expect(mockProps.playLive).toBeCalled();
  });

  it('dispatches action when clicked', () => {
    const mockStore = configureStore();
    const store = mockStore(
      fromJS({
        player: {
          liveTitle: 'mockTitle',
        },
      }),
    );

    const tree = mount(
      <Provider store={store}>
        <ConnectedHeaderPlayButton />
      </Provider>,
    );
    tree.find('.playButton').simulate('click');
    const actions = store.getActions();
    expect(actions).toEqual([playLive()]);
  });

  it('dispatches action on keyPress', () => {
    const mockStore = configureStore();
    const store = mockStore(
      fromJS({
        player: {
          liveTitle: 'mockTitle',
        },
      }),
    );
    const tree = mount(
      <Provider store={store}>
        <ConnectedHeaderPlayButton />
      </Provider>,
    );
    tree.find('.playButton').simulate('keyPress');
    const actions = store.getActions();
    expect(actions).toEqual([playLive()]);
  });
});
