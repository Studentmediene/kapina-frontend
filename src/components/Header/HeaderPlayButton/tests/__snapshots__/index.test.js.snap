// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderPlayButton /> renders correctly 1`] = `
<button
  className="container"
  onClick={[Function]}
>
  <Head>
    <title
      key="title"
    >
      Radio Revolt
    </title>
  </Head>
  <PlayPauseButton
    paused={true}
  />
  <div
    className="buttonText"
  >
    <div
      className="largeText"
    >
      Lytt direkte!
    </div>
    <div
      className="smallText"
    >
       Current show
    </div>
  </div>
</button>
`;
